geps/gep-1897.md

```diff
@@ -125,7 +141,326 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+certificates should be used.  If there are no system trusted certificates or the implementation doesn't define system
+trusted certificates, then a TLS connection must fail when the field is empty.  References to a resource in a
+different namespace are invalid.
+
+TLSConnectionPolicySpec supports the configuration of Subject Alternative Name certificates via the `AllowedSANs` field.
+One or more (up to 64) subject alternative names may be declared.  SAN certificates allow authentication with a single
+certificate that can serve multiple domain
+names that are intended to represent the same endpoint.  For example, these five SAN hosts: example.com, 
+www.example.com, example.net, IPv4 x.x.x.x, and IPv6 x.x.x.x.x.x.x.x all resolve to the same endpoint. If `AllowedSANs`
+is set, then an implementation must validate that the backend pod-provided certificate represents a SAN in its list.
+If empty, then TLS fails.  In the future, we will define more specific ways to handle the case of an empty `AllowedSANs`
+list.
+
+Finally, TLSConnectionPolicy allows the configuration of expected TLS versions via the `TLSVersions` field.
+TLS versions specify one or more (up to 3) TLS versions that the connection may use.  If the `TLSVersions` field is set,
+then the implementation must use the given versions if they are allowed.  If none of the versions are allowed, then the
+TLS connection must fail.  If no TLS versions are specified, then an implementation may choose to apply a self-determined
+default version.
+
+Thus, the following additions would be made to the Gateway API:
+
+```go
+// TLSConnectionPolicy provides a way to publish TLS configuration
+// that enables a gateway client to connect to a backend pod.
+type TLSConnectionPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of TLSConnectionPolicy.
+    Spec TLSConnectionPolicySpec `json:"spec"`
+
+    // Status defines the current state of TLSConnectionPolicy.
+    Status TLSConnectionPolicyStatus `json:"status,omitempty"`
+}
+
+// TLSConnectionPolicySpec defines the desired state of
+// TLSConnectionPolicy.
+// Note: there is no Override or Default policy configuration.
+//
+// Support: Core
+type TLSConnectionPolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // Services are the only valid API target references.
+    // Note that this config applies to the entire referenced resource
+    // by default, but this default may change in the future to provide
+    // a more granular application of the policy.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+
+    // TLS contains TLS connection policy configuration.
+    TLS *TLSConnectionPolicyConfig `json:”tls”`
+}
+
+// TLSConnectionPolicyConfig contains TLS connection policy configuration.
+type TLSConnectionPolicyConfig struct {
+    // TrustedCACertRefs contains one or more references to
```

Comment:
As discussed in the meeting, can we also add the SNI to use ? SNI is used in virtually all https connections on the internet, and allows a backend to select what cert to provide.

If we do add the SNI - could we also modify the AllowedSAN to state "if no AllowedSAN is provided, the SNI will be used to check" ( since this is also the most common scenario on the internet )

Reply:
From what I understand, the SNI has to be singleton on the TLSConnectionPolicy, and is sent by the Gateway in this process, in order for the backend application to choose which of several different certificates it would provide.  I don't think this is use case we need to support, as we want the TLSConnectionPolicy to be able to serve several endpoints.

i did not hear alot of support for adding SNI at the meeting.  i'll open a slack discussion and find out if there is widespread support, and whether TLSConnectionPolicy is the place to support it.